pyk/src/pyk/proof/reachability.py

```diff
@@ -70,6 +70,9 @@ class APRProofStep:
     module_name: str
     shortest_path_to_node: tuple[KCFG.Node, ...]
     prior_loops_cache: FrozenDict[int, tuple[int, ...]] = field(compare=False)
+    circularity: bool
+    nonzero_depth: bool
+    circularity_rule_id: str
```

Comment:
If `not circularity`, is `circularity_rule_id` relevant? If not, then perhaps the two attributes can be merged into a `str | None`.

Reply:
Honestly not sure which approach is best, but going to merge this as is.